testsuite/io/test_h5md_parallel.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+#  Copyright (C) 2021
+#      Sebastian Eibl, Max Planck Computing & Data Facility
+#
+#  This file is part of ESPResSo++.
+#
+#  ESPResSo++ is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  ESPResSo++ is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import espressopp
+import unittest
+import subprocess
+
+
+def binary_compare(f1, f2):
+    with open(f1, 'rb') as fp1, open(f2, 'rb') as fp2:
+        while True:
+            b1 = fp1.read(512)
+            b2 = fp2.read(512)
+            if b1 != b2:
+                return False
+            if not b1:
+                return True
+
+
+class TestH5MD(unittest.TestCase):
+    def test_reload_dump(self):
+        self.system, self.integrator = espressopp.standard_system.Default((10., 10., 10.))
+        self.system.rng = espressopp.esutil.RNG(42)
+        for pid in range(34):
+            pos = self.system.bc.getRandomPos()
+            self.system.storage.addParticle(pid, pos)
+        dump_h5md_parallel = espressopp.io.DumpH5MDParallel(self.system, 'reference.h5')
+        dump_h5md_parallel.dump()
+
+        self.system.storage.removeAllParticles()
+
+        restore_h5md_parallel = espressopp.io.RestoreH5MDParallel(self.system, 'reference.h5')
+        restore_h5md_parallel.restore()
+        dump_h5md_parallel = espressopp.io.DumpH5MDParallel(self.system, 'dump.h5')
+        dump_h5md_parallel.dump()
+
+        self.assertTrue(binary_compare('reference.h5', 'dump.h5'))
+
+        ls_dmp = (subprocess.run(['h5ls', '-r', 'reference.h5'], capture_output=True)).stdout
```

Comment:
Oh man, is there no way to do this internally.

Otherwise CMake will need to check for `h5ls`.

Reply:
Sure I can depend on `h5py` or `pyh5md`. I do not want to reimplement a `h5` reader.

Reply:
I now require `h5py`. I think this is the best way.